pkg/tsdb/cloudwatch/models/cloudwatch_query.go

```diff
@@ -255,19 +240,45 @@ func ParseMetricDataQueries(dataQueries []backend.DataQuery, startTime time.Time
 	}
 
 	result := make([]*CloudWatchQuery, 0, len(metricDataQueries))
+
 	for refId, mdq := range metricDataQueries {
+		alias := ""
+		if mdq.Alias != nil {
+			alias = *mdq.Alias
+		}
+
+		metricName := ""
+		if mdq.MetricName != nil {
+			metricName = *mdq.MetricName
+		}
+
+		metricQueryType := MetricQueryTypeSearch
+		if mdq.MetricQueryType != nil {
+			metricQueryType = *mdq.MetricQueryType
+		}
+
+		sqlExpression := ""
+		if mdq.SqlExpression != nil {
+			sqlExpression = *mdq.SqlExpression
+		}
+
+		expression := ""
+		if mdq.Expression != nil {
+			expression = *mdq.Expression
+		}
+
 		cwQuery := &CloudWatchQuery{
 			logger:            logger,
-			Alias:             mdq.Alias,
+			Alias:             alias,
 			RefId:             refId,
 			Id:                mdq.Id,
 			Region:            mdq.Region,
 			Namespace:         mdq.Namespace,
-			MetricName:        mdq.MetricName,
-			MetricQueryType:   mdq.MetricQueryType,
-			SqlExpression:     mdq.SqlExpression,
+			MetricName:        metricName,
+			MetricQueryType:   metricQueryType,
+			SqlExpression:     sqlExpression,
 			TimezoneUTCOffset: mdq.TimezoneUTCOffset,
-			Expression:        mdq.Expression,
+			Expression:        expression,
```

Comment:
Kind of like my other suggestion, but also is not a big deal to leave as-is.
Here you can do something like this

```
       cwQuery := &CloudWatchQuery{
			logger:            logger,
			RefId:             refId,
			Id:                mdq.Id,
			Region:            mdq.Region,
			Namespace:         mdq.Namespace,
			TimezoneUTCOffset: mdq.TimezoneUTCOffset,
		}

		if mdq.Alias != nil {
			cwQuery.Alias = *mdq.Alias
		}

... and so on for the other fields
```

Reply:
thanks, this looks better. i made the updates, the only one i kept was this

```go
		metricQueryType := MetricQueryTypeSearch
		if mdq.MetricQueryType != nil {
			metricQueryType = *mdq.MetricQueryType
		}
```

since this is defaulting to `MetricQueryTypeSearch`. any preference to this or setting it like this

```go
		cwQuery := &CloudWatchQuery{
			...
			MetricQueryType:   MetricQueryTypeSearch,
			...
		}

		if mdq.MetricQueryType != nil {
			cwQuery.MetricQueryType = *mdq.MetricQueryType
		}
```

or something else?

Reply:
Any reason why we wouldn't do the same thing as the others?
In the original code was it explicitly set to default as type search or was it also just a fallback to 0?

Reply:
> Any reason why we wouldn't do the same thing as the others? In the original code was it explicitly set to default as type search or was it also just a fallback to 0?

ah, you're right. i think i hallucinated it being explicitly set.